pkg/noun/events.c

```diff
@@ -988,7 +1051,7 @@ _ce_loom_mapf_north(c3_i fid_i, c3_w pgs_w, c3_w old_w)
     if ( MAP_FAILED == mmap(_ce_ptr(0),
                             _ce_len(pgs_w),
                             PROT_READ,
-                            (MAP_FIXED | MAP_PRIVATE),
+                            (MAP_FIXED | MAP_SHARED),
```

Comment:
`MAP_SHARED` here breaks the atomicity of snapshot updates, allowing the kernel to flush modifications to disk unpredictably. What motivated the change?

Reply:
It occurs to me that another way to trade off disk/memory usage would be to double-buffer north.bin on disk, which would allow it to always be read-write/MAP_SHARED. But that would be a more invasive change.

Reply:
Oh, of course.  I was not seeing the memory numbers I expected, so I changed this, but I'm pretty sure that was because the ephemeral space wasn't yet getting fully mapped in.  I'll change it back and see if the memory numbers look the same (I think they will).

Technically, it probably doesn't make a difference because `PROT_READ` should mean we never modify those addresses while they're mapped to file anyway, but `MAP_PRIVATE` is more correct.

Reply:
You'd still get the fault in userspace, but without CoW semantics in the kernel.